packages/realm-react/src/useEmailPasswordAuth.tsx

```diff
@@ -0,0 +1,235 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { useCallback, useEffect, useState } from "react";
+import { useApp } from "./AppProvider";
+import { AuthError, AuthResult, OperationState } from "./types";
+import { Realm, App, Object, User, Credentials } from "realm";
+
+interface UseEmailPasswordAuth {
+  /**
+   * Convenience function to login a user with an email and password - users
+   * could also call `logIn(Realm.Credentials.emailPassword(email, password)).
+   *
+   * TODO: Does it make sense to have this convenience function? Should we add
```

Comment:
I think it makes sense as we are saving the user from writing some trivial code.